src/block-management/create-block-and-log-streamer.js

```diff
@@ -52,32 +61,33 @@ function createBlockAndLogStreamer(configuration, transport, callback) {
       subscribeToNewHeads: transport.subscribeToNewHeads,
       unsubscribeFromNewHeads: transport.unsubscribeFromNewHeads
     }, configuration.pollingIntervalMilliseconds);
-    var blockAndLogStreamer = BlockAndLogStreamer.createCallbackStyle(transport.getBlockByHash, transport.getLogs, {
-      blockRetention: configuration.blockRetention
-    });
-
-    internalState.setState({ blockAndLogStreamer: blockAndLogStreamer, blockNotifier: blockNotifier });
-
-    callback = isFunction(callback) ? callback : function (e) { if (e) console.log(e); };
-    function subscribeToBlockNotifier() {
-      blockNotifier.subscribe(function (block) {
-        blockAndLogStreamer.reconcileNewBlockCallbackStyle(block, function (err) { if (err) return console.error(err); });
+    var blockAndLogStreamer = new BlockAndLogStreamer(function (hash) {
+      return new Promise(function (resolve, reject) {
+        transport.getBlockByHash(hash, function (err, block) {
+          if (err) throw err;
```

Comment:
I believe this should be:
```typescript
if (err) reject(err);
else resolve(block);`
```

Reply:
If you aren't using async/await, you generally don't want to ever `throw`, because JS is terrible.

Reply:
Good catch (no pun intended)